### DISCUSSION TITLE
Wie kann ich einem Meta-Info Feld vom Typ "select" eine "Leer"-Option hinzufügen?

### DISCUSSION THREAD

**Discussion by tyrant88** [FriendsOfREDAXO/.github#43] [Q&A] [answered]
Ich habe ein Meta-Info Feld vom Typ "select" mit einer SQL-Query angelegt.
`SELECT name,id FROM rex_product_groups WHERE status='A'`
Jetzt möchte ich aber eine "Leer"-Option davor haben. 
Wie ging das nochmal am Schönsten?

**Comment by christophboecker** [comment, marked as answer]
Die Lösung ist hier in den Tricks zu finden: https://friendsofredaxo.github.io/tricks/snippets/queries#option-please-select

**Reply by tyrant88** [reply]
Ah, also mit UNION!
Hab es nur nach vorne gezogen:
`SELECT ' Bitte wählen' AS name, "" AS id UNION SELECT name, filename AS id FROM rex_project_icon ORDER BY name`

**Reply by christophboecker** [reply]
> Hab es nur nach vorne gezogen

Falls es um die Sortierung geht: das hat keinen Einfluß. `order by` wirkt auf das UNION-Ergebnis. Man kriegt das "Bitte wählen" durch das Leerzeichen vor "Bitte" oder ein anderes niedriges Zeichen nach vorne ("(Bitte wählen)", "--Bitte wählen--" o.ä.)

**Reply by tyrant88** [reply]
Ich geb zu, das "order by" hab ich weggelassen, deshalb macht es was ich will :-)

**Reply by tyrant88** [reply]
ne Klammer um das zweite SELECT mit dem order by könnte es auch machen, oder?

**Reply by christophboecker** [reply]
> ne Klammer um das zweite SELECT mit dem order by könnte es auch machen, oder?

:-) leider nein, hab´s ausprobiert.

**Reply by tyrant88** [reply]
Also das:
`SELECT 'Bitte wählen' AS name, "" AS id UNION (SELECT name, id AS id FROM rex_product_groups ORDER BY name)`
funktioniert bei mir.

**Reply by christophboecker** [reply]
Ja stimmt, ich hatte nen (Denk-)Fehler drin. Ja klar. so ist super.
